[2024-09-13 00:12:12,574] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
/workspace/Oobleck/oobleck/__init__.py
node_index: 16
[2024-09-13 00:12:15,654] [WARNING] [agent.py:154:_launch_workers] Profile data for model gpt2 not found. Exception Error parsing json file: /workspace/Oobleck/tmp/profiles/gpt3_2_7B-8-17-1/mb8.json Launching profiler...
[2024-09-13 00:12:17,448] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 00:12:20,504] [INFO] [profiler.py:270:profile] Profiling model gpt2. rank 16, world_size 17, num_workers_per_node 1.
Dataset: model_name: gpt2, dataset_path: wikitext, dataset_name: wikitext-2-raw-v1, max_seq_length: 1024
[2024-09-13 00:12:22,464] [INFO] [profiler.py:276:profile] Dataset loaded
model_name: gpt2, model_tag: gpt3_2_7B
[2024-09-13 00:12:23,630] [INFO] [profiler.py:284:profile] model loaded
[2024-09-13 00:12:53,198] [INFO] [profiler.py:292:profile] init pg. master_addr: 172.21.0.42, master_port: 23456
[2024-09-13 00:13:08,240] [INFO] [profiler.py:306:profile] Profiling model execution latency.
[2024-09-13 00:13:08,241] [INFO] [profiler.py:112:profile_execution_layers] before barrier
[2024-09-13 00:13:20,611] [INFO] [profiler.py:114:profile_execution_layers] after barrier
[2024-09-13 00:13:20,611] [INFO] [profiler.py:120:profile_execution_layers] before broadcast
[2024-09-13 00:13:20,611] [INFO] [profiler.py:122:profile_execution_layers] after broadcast
[2024-09-13 00:13:20,618] [INFO] [profiler.py:317:profile] Profiling cross-node allreduce latency.
[2024-09-13 00:13:20,618] [INFO] [profiler.py:160:profile_allreduce_across_nodes] Profile allreduce across 17 nodes latency
[2024-09-13 00:13:20,905] [INFO] [profiler.py:326:profile] Profiling in-node allreduce latency.
[2024-09-13 00:13:20,905] [INFO] [profiler.py:209:profile_allreduce_in_node] Profile allreduce within a node latency
[2024-09-13 00:13:21,941] [INFO] [agent.py:189:_launch_workers] in agent. my_ip_port 172.21.0.91:2220, node_ip_ports ['172.21.0.42:2220', '172.21.0.42:2221', '172.21.0.42:2222', '172.21.0.42:2223', '172.21.0.46:2220', '172.21.0.46:2221', '172.21.0.46:2222', '172.21.0.46:2223', '172.21.0.47:2220', '172.21.0.47:2221', '172.21.0.47:2222', '172.21.0.47:2223', '172.21.0.90:2220', '172.21.0.90:2221', '172.21.0.90:2222', '172.21.0.90:2223', '172.21.0.91:2220']
[2024-09-13 00:13:21,942] [INFO] [agent.py:192:_launch_workers] Launching worker 0...
[2024-09-13 00:13:23,783] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 00:13:26,835] [INFO] [worker.py:23:worker_main] Initializing Oobleck Engine...
[2024-09-13 00:13:26,835] [INFO] [worker.py:24:worker_main] in worker main: my_ip 172.21.0.91:2220
/root/miniconda3/envs/oobleck/lib/python3.10/site-packages/transformers/training_args.py:1281: FutureWarning: using `no_cuda` is deprecated and will be removed in version 5.0 of ðŸ¤— Transformers. Use `use_cpu` instead
  warnings.warn(
Dataset: model_name: gpt2, dataset_path: wikitext, dataset_name: wikitext-2-raw-v1, max_seq_length: 1024
[2024-09-13 00:13:28,710] [INFO] [engine.py:635:_initialize_engine] model arguments: {'n_embd': 2560, 'n_head': 32, 'n_positions': 1024, 'num_hidden_layers': 32}
model_name: gpt2, model_tag: gpt3_2_7B
mb size: 34, ar across: 34, ar in: 34
Loading done. creating layer execution results...
Returning from get_profiler_results
[2024-09-13 00:13:29,857] [INFO] [engine.py:677:_initialize_engine] Number of nodes range: (3, 17)
min_nodes: 3, max_nodes: 17, gpus: 1, 
Create pipeline templates
StageExecutionResult[0:12] with 1 devices
StageExecutionResult[13:23] with 1 devices
StageExecutionResult[24:33] with 1 devices
StageExecutionResult[0:9] with 1 devices
StageExecutionResult[10:18] with 1 devices
StageExecutionResult[19:26] with 1 devices
StageExecutionResult[27:33] with 1 devices
StageExecutionResult[0:7] with 1 devices
StageExecutionResult[8:14] with 1 devices
StageExecutionResult[15:21] with 1 devices
StageExecutionResult[22:27] with 1 devices
StageExecutionResult[28:33] with 1 devices
StageExecutionResult[0:6] with 1 devices
StageExecutionResult[7:12] with 1 devices
StageExecutionResult[13:18] with 1 devices
StageExecutionResult[19:23] with 1 devices
StageExecutionResult[24:28] with 1 devices
StageExecutionResult[29:33] with 1 devices
StageExecutionResult[0:5] with 1 devices
StageExecutionResult[6:10] with 1 devices
StageExecutionResult[11:15] with 1 devices
StageExecutionResult[16:20] with 1 devices
StageExecutionResult[21:25] with 1 devices
StageExecutionResult[26:29] with 1 devices
StageExecutionResult[30:33] with 1 devices
StageExecutionResult[0:5] with 1 devices
StageExecutionResult[6:10] with 1 devices
StageExecutionResult[11:14] with 1 devices
StageExecutionResult[15:18] with 1 devices
StageExecutionResult[19:22] with 1 devices
StageExecutionResult[23:26] with 1 devices
StageExecutionResult[27:30] with 1 devices
StageExecutionResult[31:33] with 1 devices
StageExecutionResult[0:4] with 1 devices
StageExecutionResult[5:8] with 1 devices
StageExecutionResult[9:12] with 1 devices
StageExecutionResult[13:16] with 1 devices
StageExecutionResult[17:20] with 1 devices
StageExecutionResult[21:24] with 1 devices
StageExecutionResult[25:27] with 1 devices
StageExecutionResult[28:30] with 1 devices
StageExecutionResult[31:33] with 1 devices
StageExecutionResult[0:4] with 1 devices
StageExecutionResult[5:8] with 1 devices
StageExecutionResult[9:12] with 1 devices
StageExecutionResult[13:15] with 1 devices
StageExecutionResult[16:18] with 1 devices
StageExecutionResult[19:21] with 1 devices
StageExecutionResult[22:24] with 1 devices
StageExecutionResult[25:27] with 1 devices
StageExecutionResult[28:30] with 1 devices
StageExecutionResult[31:33] with 1 devices
StageExecutionResult[0:4] with 1 devices
StageExecutionResult[5:7] with 1 devices
StageExecutionResult[8:10] with 1 devices
StageExecutionResult[11:13] with 1 devices
StageExecutionResult[14:16] with 1 devices
StageExecutionResult[17:19] with 1 devices
StageExecutionResult[20:22] with 1 devices
StageExecutionResult[23:25] with 1 devices
StageExecutionResult[26:28] with 1 devices
StageExecutionResult[29:31] with 1 devices
StageExecutionResult[32:33] with 1 devices
StageExecutionResult[0:3] with 1 devices
StageExecutionResult[4:6] with 1 devices
StageExecutionResult[7:9] with 1 devices
StageExecutionResult[10:12] with 1 devices
StageExecutionResult[13:15] with 1 devices
StageExecutionResult[16:18] with 1 devices
StageExecutionResult[19:21] with 1 devices
StageExecutionResult[22:24] with 1 devices
StageExecutionResult[25:27] with 1 devices
StageExecutionResult[28:30] with 1 devices
StageExecutionResult[31:32] with 1 devices
StageExecutionResult[33:33] with 1 devices
StageExecutionResult[0:3] with 1 devices
StageExecutionResult[4:6] with 1 devices
StageExecutionResult[7:9] with 1 devices
StageExecutionResult[10:12] with 1 devices
StageExecutionResult[13:15] with 1 devices
StageExecutionResult[16:18] with 1 devices
StageExecutionResult[19:21] with 1 devices
StageExecutionResult[22:24] with 1 devices
StageExecutionResult[25:26] with 1 devices
StageExecutionResult[27:28] with 1 devices
StageExecutionResult[29:30] with 1 devices
StageExecutionResult[31:32] with 1 devices
StageExecutionResult[33:33] with 1 devices
StageExecutionResult[0:3] with 1 devices
StageExecutionResult[4:6] with 1 devices
StageExecutionResult[7:9] with 1 devices
StageExecutionResult[10:12] with 1 devices
StageExecutionResult[13:15] with 1 devices
StageExecutionResult[16:17] with 1 devices
StageExecutionResult[18:19] with 1 devices
StageExecutionResult[20:21] with 1 devices
StageExecutionResult[22:23] with 1 devices
StageExecutionResult[24:25] with 1 devices
StageExecutionResult[26:27] with 1 devices
StageExecutionResult[28:29] with 1 devices
StageExecutionResult[30:31] with 1 devices
StageExecutionResult[32:33] with 1 devices
StageExecutionResult[0:3] with 1 devices
StageExecutionResult[4:6] with 1 devices
StageExecutionResult[7:8] with 1 devices
StageExecutionResult[9:11] with 1 devices
StageExecutionResult[12:14] with 1 devices
StageExecutionResult[15:16] with 1 devices
StageExecutionResult[17:18] with 1 devices
StageExecutionResult[19:20] with 1 devices
StageExecutionResult[21:22] with 1 devices
StageExecutionResult[23:24] with 1 devices
StageExecutionResult[25:26] with 1 devices
StageExecutionResult[27:28] with 1 devices
StageExecutionResult[29:30] with 1 devices
StageExecutionResult[31:32] with 1 devices
StageExecutionResult[33:33] with 1 devices
StageExecutionResult[0:3] with 1 devices
StageExecutionResult[4:6] with 1 devices
StageExecutionResult[7:8] with 1 devices
StageExecutionResult[9:10] with 1 devices
StageExecutionResult[11:12] with 1 devices
StageExecutionResult[13:14] with 1 devices
StageExecutionResult[15:16] with 1 devices
StageExecutionResult[17:18] with 1 devices
StageExecutionResult[19:20] with 1 devices
StageExecutionResult[21:22] with 1 devices
StageExecutionResult[23:24] with 1 devices
StageExecutionResult[25:26] with 1 devices
StageExecutionResult[27:28] with 1 devices
StageExecutionResult[29:30] with 1 devices
StageExecutionResult[31:32] with 1 devices
StageExecutionResult[33:33] with 1 devices
StageExecutionResult[0:2] with 1 devices
StageExecutionResult[3:4] with 1 devices
StageExecutionResult[5:6] with 1 devices
StageExecutionResult[7:8] with 1 devices
StageExecutionResult[9:10] with 1 devices
StageExecutionResult[11:12] with 1 devices
StageExecutionResult[13:14] with 1 devices
StageExecutionResult[15:16] with 1 devices
StageExecutionResult[17:18] with 1 devices
StageExecutionResult[19:20] with 1 devices
StageExecutionResult[21:22] with 1 devices
StageExecutionResult[23:24] with 1 devices
StageExecutionResult[25:26] with 1 devices
StageExecutionResult[27:28] with 1 devices
StageExecutionResult[29:30] with 1 devices
StageExecutionResult[31:32] with 1 devices
StageExecutionResult[33:33] with 1 devices
[2024-09-13 00:13:38,297] [INFO] [worker.py:27:worker_main] Initializing torch.distributed...
[2024-09-13 00:13:38,297] [DEBUG] [engine.py:725:initialize_distributed] regenerate rank_map {'172.21.0.42:2220': [0], '172.21.0.42:2221': [1], '172.21.0.42:2222': [2], '172.21.0.42:2223': [3], '172.21.0.46:2220': [4], '172.21.0.46:2221': [5], '172.21.0.46:2222': [6], '172.21.0.46:2223': [7], '172.21.0.47:2220': [8], '172.21.0.47:2221': [9], '172.21.0.47:2222': [10], '172.21.0.47:2223': [11], '172.21.0.90:2220': [12], '172.21.0.90:2221': [13], '172.21.0.90:2222': [14], '172.21.0.90:2223': [15], '172.21.0.91:2220': [16]}
[2024-09-13 00:13:38,298] [INFO] [engine.py:739:initialize_distributed] init pg: rank 16, world_size: 17, rank_map: {'172.21.0.42:2220': [0], '172.21.0.42:2221': [1], '172.21.0.42:2222': [2], '172.21.0.42:2223': [3], '172.21.0.46:2220': [4], '172.21.0.46:2221': [5], '172.21.0.46:2222': [6], '172.21.0.46:2223': [7], '172.21.0.47:2220': [8], '172.21.0.47:2221': [9], '172.21.0.47:2222': [10], '172.21.0.47:2223': [11], '172.21.0.90:2220': [12], '172.21.0.90:2221': [13], '172.21.0.90:2222': [14], '172.21.0.90:2223': [15], '172.21.0.91:2220': [16]}
[2024-09-13 00:13:38,298] [INFO] [engine.py:755:initialize_distributed] Waiting for a port information...
[2024-09-13 00:13:52,220] [DEBUG] [agent.py:298:on_receive_response] Receiving: (<Response.FORWARD_RANK0_PORT: 5>, <RequestType.UNDEFINED: 0>)
[2024-09-13 00:13:52,221] [DEBUG] [agent.py:243:on_receive_worker_port] agent recv TCP Store port 38329 from master
[2024-09-13 00:13:52,221] [INFO] [engine.py:758:initialize_distributed] Received torch master: 172.21.0.42:2220.38329
[2024-09-13 00:13:52,235] [INFO] [comm.py:637:init_distributed] cdb=None
[2024-09-13 00:13:52,236] [INFO] [engine.py:775:initialize_distributed] [rank: 16] Distributed initialization is done.
[2024-09-13 00:13:52,236] [INFO] [worker.py:34:worker_main] Instantiating pipelines...
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:52,580] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:52,893] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:53,212] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value 2.5059E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:53,818] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:54,181] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:54,537] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:54,893] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:55,253] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:55,528] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value 2.5059E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:56,088] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4942E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:56,982] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4942E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:13:58,176] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value -7.4942E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:14:00,169] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:14:01,372] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:14:03,647] [INFO] [instantiator.py:256:get_best_execution_plan] Best execution plan: 3 x <oobleck.PipelineTemplate.3nodes> pipelines (b: 23)
1 x <oobleck.PipelineTemplate.8nodes> pipelines (b: 59)
B: 128
dataloader: mbs 8
[2024-09-13 00:14:22,279] [INFO] [engine.py:159:_reconfiguration_listener_fn] ReconfigureEngine: start reconfigure listening
[2024-09-13 00:14:22,280] [INFO] [worker.py:36:worker_main] Begin training...
step: 0
step in try: 0
load_mbs: 0.0007686614990234375, forward time: 7.26430869102478, backward time: 14.294102668762207
load_mbs cnt: 59, forward cnt: 59, backward cnt: 59
Process SpawnProcess-2:
Traceback (most recent call last):
  File "/root/miniconda3/envs/oobleck/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/root/miniconda3/envs/oobleck/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/workspace/Oobleck/oobleck/elastic/worker.py", line 37, in worker_main
    engine.train()
  File "/workspace/Oobleck/oobleck/execution/engine.py", line 895, in train
    self._train_step()
  File "/workspace/Oobleck/oobleck/utils/timer.py", line 15, in wrapper
    result = func(s, *args, **kwargs)
  File "/workspace/Oobleck/oobleck/execution/engine.py", line 841, in _train_step
    self._dp_engine.do_allreduce()
  File "/workspace/Oobleck/oobleck/execution/engine.py", line 550, in do_allreduce
    layer.reduce_gradients(process_groups)
  File "/workspace/Oobleck/oobleck/execution/layer.py", line 305, in reduce_gradients
    torch.distributed.all_reduce(tensor=grad, group=process_group)
  File "/root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py", line 1436, in wrapper
    return func(*args, **kwargs)
  File "/root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py", line 1687, in all_reduce
    work = group.allreduce([tensor], opts)
RuntimeError: [3] is setting up NCCL communicator and retrieving ncclUniqueId from [0] via c10d key-value store by key '0', but store->get('0') got error: Socket Timeout
Exception raised from recvBytes at /opt/conda/conda-bld/pytorch_1678411187366/work/torch/csrc/distributed/c10d/Utils.hpp:604 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7fb6589ae4d7 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, char const*) + 0x68 (0x7fb658978434 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libc10.so)
frame #2: c10d::TCPStore::doWait(c10::ArrayRef<std::string>, std::chrono::duration<long, std::ratio<1l, 1000l> >) + 0xd8 (0x7fb69c1e5248 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #3: c10d::TCPStore::doGet(std::string const&) + 0x22 (0x7fb69c1e5ef2 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #4: c10d::TCPStore::get(std::string const&) + 0x59 (0x7fb69c1e5f79 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #5: c10d::PrefixStore::get(std::string const&) + 0x31 (0x7fb69c1a5441 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #6: c10d::PrefixStore::get(std::string const&) + 0x31 (0x7fb69c1a5441 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #7: c10d::PrefixStore::get(std::string const&) + 0x31 (0x7fb69c1a5441 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #8: c10d::ProcessGroupNCCL::broadcastUniqueNCCLID(ncclUniqueId*, bool, std::string const&, int) + 0xaf (0x7fb6599706ff in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so)
frame #9: c10d::ProcessGroupNCCL::getNCCLComm(std::string const&, std::vector<c10::Device, std::allocator<c10::Device> > const&, c10d::OpType, int, bool) + 0x201 (0x7fb6599743b1 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so)
frame #10: <unknown function> + 0xf034dd (0x7fb65997b4dd in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so)
frame #11: c10d::ProcessGroupNCCL::allreduce_impl(std::vector<at::Tensor, std::allocator<at::Tensor> >&, c10d::AllreduceOptions const&) + 0x21 (0x7fb65997c8e1 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so)
frame #12: c10d::ProcessGroupNCCL::allreduce(std::vector<at::Tensor, std::allocator<at::Tensor> >&, c10d::AllreduceOptions const&) + 0x39d (0x7fb65997f59d in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so)
frame #13: <unknown function> + 0x501d534 (0x7fb69c19a534 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #14: <unknown function> + 0x502009f (0x7fb69c19d09f in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #15: <unknown function> + 0x50399b3 (0x7fb69c1b69b3 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #16: <unknown function> + 0xb6d581 (0x7fb6a5d1c581 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_python.so)
frame #17: <unknown function> + 0x3b7265 (0x7fb6a5566265 in /root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/lib/libtorch_python.so)
frame #18: <unknown function> + 0x14bb15 (0x55c5ebb88b15 in /root/miniconda3/envs/oobleck/bin/python)
frame #19: _PyObject_MakeTpCall + 0x152 (0x55c5ebb7fa62 in /root/miniconda3/envs/oobleck/bin/python)
frame #20: <unknown function> + 0xe446c (0x55c5ebb2146c in /root/miniconda3/envs/oobleck/bin/python)
frame #21: _PyEval_EvalFrameDefault + 0x49c9 (0x55c5ebc25439 in /root/miniconda3/envs/oobleck/bin/python)
frame #22: _PyFunction_Vectorcall + 0x25d (0x55c5ebc031bd in /root/miniconda3/envs/oobleck/bin/python)
frame #23: PyObject_Call + 0xb8 (0x55c5ebb86028 in /root/miniconda3/envs/oobleck/bin/python)
frame #24: _PyEval_EvalFrameDefault + 0x2c0b (0x55c5ebc2367b in /root/miniconda3/envs/oobleck/bin/python)
frame #25: _PyFunction_Vectorcall + 0x798 (0x55c5ebc036f8 in /root/miniconda3/envs/oobleck/bin/python)
frame #26: _PyEval_EvalFrameDefault + 0x125a (0x55c5ebc21cca in /root/miniconda3/envs/oobleck/bin/python)
frame #27: <unknown function> + 0x1c73a5 (0x55c5ebc043a5 in /root/miniconda3/envs/oobleck/bin/python)
frame #28: _PyEval_EvalFrameDefault + 0x49c9 (0x55c5ebc25439 in /root/miniconda3/envs/oobleck/bin/python)
frame #29: _PyFunction_Vectorcall + 0x25d (0x55c5ebc031bd in /root/miniconda3/envs/oobleck/bin/python)
frame #30: _PyEval_EvalFrameDefault + 0x60b (0x55c5ebc2107b in /root/miniconda3/envs/oobleck/bin/python)
frame #31: _PyFunction_Vectorcall + 0x25d (0x55c5ebc031bd in /root/miniconda3/envs/oobleck/bin/python)
frame #32: PyObject_Call + 0x1aa (0x55c5ebb8611a in /root/miniconda3/envs/oobleck/bin/python)
frame #33: _PyEval_EvalFrameDefault + 0x2c0b (0x55c5ebc2367b in /root/miniconda3/envs/oobleck/bin/python)
frame #34: _PyFunction_Vectorcall + 0x9eb (0x55c5ebc0394b in /root/miniconda3/envs/oobleck/bin/python)
frame #35: _PyEval_EvalFrameDefault + 0x60b (0x55c5ebc2107b in /root/miniconda3/envs/oobleck/bin/python)
frame #36: _PyFunction_Vectorcall + 0x25d (0x55c5ebc031bd in /root/miniconda3/envs/oobleck/bin/python)
frame #37: _PyEval_EvalFrameDefault + 0x60b (0x55c5ebc2107b in /root/miniconda3/envs/oobleck/bin/python)
frame #38: _PyFunction_Vectorcall + 0x25d (0x55c5ebc031bd in /root/miniconda3/envs/oobleck/bin/python)
frame #39: PyObject_Call + 0x1aa (0x55c5ebb8611a in /root/miniconda3/envs/oobleck/bin/python)
frame #40: _PyEval_EvalFrameDefault + 0x2c0b (0x55c5ebc2367b in /root/miniconda3/envs/oobleck/bin/python)
frame #41: _PyFunction_Vectorcall + 0x25d (0x55c5ebc031bd in /root/miniconda3/envs/oobleck/bin/python)
frame #42: _PyEval_EvalFrameDefault + 0x60b (0x55c5ebc2107b in /root/miniconda3/envs/oobleck/bin/python)
frame #43: _PyFunction_Vectorcall + 0x25d (0x55c5ebc031bd in /root/miniconda3/envs/oobleck/bin/python)
frame #44: _PyEval_EvalFrameDefault + 0x60b (0x55c5ebc2107b in /root/miniconda3/envs/oobleck/bin/python)
frame #45: _PyFunction_Vectorcall + 0x25d (0x55c5ebc031bd in /root/miniconda3/envs/oobleck/bin/python)
frame #46: _PyEval_EvalFrameDefault + 0x304 (0x55c5ebc20d74 in /root/miniconda3/envs/oobleck/bin/python)
frame #47: _PyFunction_Vectorcall + 0x25d (0x55c5ebc031bd in /root/miniconda3/envs/oobleck/bin/python)
frame #48: _PyEval_EvalFrameDefault + 0x125a (0x55c5ebc21cca in /root/miniconda3/envs/oobleck/bin/python)
frame #49: <unknown function> + 0x1c51b9 (0x55c5ebc021b9 in /root/miniconda3/envs/oobleck/bin/python)
frame #50: PyEval_EvalCode + 0x87 (0x55c5ebcb5f67 in /root/miniconda3/envs/oobleck/bin/python)
frame #51: <unknown function> + 0x279029 (0x55c5ebcb6029 in /root/miniconda3/envs/oobleck/bin/python)
frame #52: <unknown function> + 0x29ec94 (0x55c5ebcdbc94 in /root/miniconda3/envs/oobleck/bin/python)
frame #53: PyRun_StringFlags + 0x7d (0x55c5ebce2eed in /root/miniconda3/envs/oobleck/bin/python)
frame #54: PyRun_SimpleStringFlags + 0x3d (0x55c5ebce2f4d in /root/miniconda3/envs/oobleck/bin/python)
frame #55: Py_RunMain + 0x26c (0x55c5ebce31ec in /root/miniconda3/envs/oobleck/bin/python)
frame #56: Py_BytesMain + 0x39 (0x55c5ebce3469 in /root/miniconda3/envs/oobleck/bin/python)
frame #57: __libc_start_main + 0xe7 (0x7fb77d9e6c87 in /lib/x86_64-linux-gnu/libc.so.6)
frame #58: <unknown function> + 0x2112d1 (0x55c5ebc4e2d1 in /root/miniconda3/envs/oobleck/bin/python)
. This may indicate a possible application crash on rank 0 or a network set up issue.
